front-end/src/screens/Settings/address.tsx

```diff
@@ -183,74 +175,117 @@ const Address = ({ className }: Props): JSX.Element => {
 		);
 	}
 
-	if (accounts.length === 0) {
-		return (
-			<Form className={className} standalone={false}>
-				<Form.Group>
-					<Form.Field width={16}>
-						<label>Linked address</label>
-						<div className='text-muted'>Associate your account with an on chain address using the <a href={getExtensionUrl()}>Polkadot-js extension</a>.</div>
-						<div className='link-button-container'>
-							<Button primary onClick={handleDetect}>
-								{currentUser?.addresses?.length ? 'Show linked addresses' : 'Link address'}
-							</Button>
-						</div>
-					</Form.Field>
-				</Form.Group>
-			</Form>
-		);
-	}
-
 	return (
 		<Form className={className} standalone={false}>
-			<Form.Group>
-				<Form.Field width={16}>
-					<label className='header'>Available addresses</label>
-					<div className='ui list'>
-						{accounts.map(account => {
-							const address = getEncodedAddress(account.address);
+			{accounts.length === 0 ?
+				<>
+					{currentUser?.addresses?.length ? <Form.Group>
```

Comment:
The 2 cases have very similar code (whether the accounts come from the extension or from the userContext). We should combine this in one function, it'll make things much easier to read/debug/maintain.